engine-types/src/types.rs

```diff
@@ -183,6 +295,10 @@ impl Wei {
     pub fn checked_add(self, rhs: Self) -> Option<Self> {
         self.0.checked_add(rhs.0).map(Self)
     }
+
+    pub fn into_u128(self) -> u128 {
+        self.0.as_u128()
+    }
```

Comment:
Good point for discussion. Before that, we used `as_u128` instead. When it's overflow it will panics.
There is really long path for a lot of core functions without Result type.
So we can use `low_u128`, but it doesn't check overflow.
Or introduce Result type for a lot of core Engine functions.

Reply:
Is it that much work to introduce `Result`? Lots of functions already return `Result` now after the recent refactors, so hopefully adding a `Result` here would only bubble up to being a new error variant on some functions that are already returning errors.

Reply:
I spent some time analyzing where we can use it. The problem is in point related to the Balance key prefix. We use the same prefix for `u128` and `U256` balances. But I found a solution for that. It was committed.